pkg/cache/nodes.go

```diff
@@ -100,10 +101,18 @@ func (nc *schedulerNodes) addAndReportNode(node *v1.Node, reportNode bool) {
 
 	// add node to nodes map
 	if _, ok := nc.nodesMap[node.Name]; !ok {
+		nodeLabels, err := json.Marshal(node.Labels)
+		if err != nil {
+			log.Logger().Error("failed to marshall node labels to json", zap.Error(err))
+			return
```

Comment:
I don't think we should return here, even we can't marshall node labels, we should still report the node
otherwise the node will be skipped adding to the scheduler, that might not be something we want

Reply:
Yeah, good catch. 

Since we are only using node labels to support node affinity with a **_preferred_** scheduling policy, it makes sense to not skip this node even if labels can't be marshaled.